File: SuboptimalProcedureLearner/SuboptimalProcedureLearner/SuboptimalProcedureLearner/OperatorInstance.cs
using System;
using System.Collections.Generic;

using Datastructures;

namespace SuboptimalProcedureLearner.SuboptimalProcedureLearner
{
    class OperatorInstance
    {
        public OperatorInstance(Operator correspondingOperator)
        {
            this.correspondingOperator = correspondingOperator;
        }

        // used by the Executive to query informations(wiring, dependencies, etc)
        public Operator getCorrespondingOperator()
        {
            return correspondingOperator;
        }

        public OperatorInstance[] calleeOperatorInstances;
        public List<Variadic> calleeResults = new List<Variadic>();

        public int operationState = 0;

        private Operator correspondingOperator;

        public Variadic[] constants;
    }
}

File: src/java/ptrman/agi-x0/src/java/SuboptimalProcedureLearner/OperatorInstance.java
package SuboptimalProcedureLearner;

import Datastructures.Variadic;

import java.util.ArrayList;
import java.util.List;

public class OperatorInstance {
    public OperatorInstance(AbstractOperatorBase correspondingOperator) {
        this.correspondingOperator = correspondingOperator;
    }

    // used by the Executive to query informations(wiring, dependencies, etc)
    public AbstractOperatorBase getCorrespondingOperator() {
        return correspondingOperator;
    }

    public OperatorInstance[] calleeOperatorInstances;
    public List<Variadic> calleeResults = new ArrayList<>();
    public int operationState = 0;
    private AbstractOperatorBase correspondingOperator;
    public Variadic[] constants;
}
